crates/agglayer-clock/src/block.rs

```diff
@@ -151,13 +152,16 @@ where
             .await
             .map_err(|_| BlockClockError::SubscribeBlocks)?;
 
-        let current_l1_block = current_l1_block.as_u64();
-        if current_l1_block < self.genesis_block {
-            return Err(BlockClockError::GenesisBlockNumber(
-                current_l1_block,
-                self.genesis_block,
-            ));
-        };
+        let mut current_l1_block = current_l1_block.as_u64();
+        while !current_l1_block < self.genesis_block {
```

Comment:
not sure of why the negation?

there is also the first comment at the beginning of this function to adapt

```diff
        // Start by setting the current Block height based on the current L1 Block
        // number. If the current L1 Block number is less than the genesis block
-       // number, we exit the process with an println message.
+       // number, we walk the L1 block stream until reaching the genesis block.
```